src/sawtooth/Application/SmallBankBatchBuilder.js

```diff
@@ -0,0 +1,363 @@
+        let address = familyNameSpace + hexAddress.substring(0,64);
+        return address;
+    }
+
+    /**
+     * It is same as calculateAddress but calculates multiple address at a time.
+     * and 6 character from account number.
+     * @param {number} args array of account numbers
+     * @returns {string} list valid state addresses
+     */
+    calculateAddresses(args) {
+        let addresses = [];
+        for (let key in args) {
+            let address = this.calculateAddress(key);
+            addresses.push(address);
+        }
+        return addresses;
+    }
+
+    /**
+     * Create the sawtooth transaction from small bank payload
+     * @param {object} args small bank payload
+     * @param {object} signer used for signing transaction
+     * @returns {object} transaction object
+     */
+    buildTransaction(args, signer) {
+        const {createHash} = require('crypto');
+        const {protobuf} = require('sawtooth-sdk');
+
+        //get customer ids from the argument and calculate the addresses from
+        //customer ids
+        let ids = getCustomerIds(args);
+        let in_address = this.calculateAddress(ids[0]);
+        let input_address=[in_address];
+        let output_address=[in_address];
+        if(ids.length === 2) {
+            let out_address = this.calculateAddress(ids[1]);
+            input_address.push(out_address);
+            output_address.push(out_address);
+        }
+        //Generate protobuf payload from input args
+        const payloadBytes = buildPayload(args);
+
+        //Construct transaction header
+        const transactionHeaderBytes = protobuf.TransactionHeader.encode({
+            familyName: this.familyName,
+            familyVersion: this.familyVersion,
+            inputs: input_address,
+            outputs: output_address,
+            signerPublicKey: signer.getPublicKey().asHex(),
+            batcherPublicKey: signer.getPublicKey().asHex(),
+            dependencies: [],
+            payloadSha512: createHash('sha512').update(payloadBytes).digest('hex')
+        }).finish();
+
+        //construct transaction
+        const txnSignature = signer.sign(transactionHeaderBytes);
+        const transaction = protobuf.Transaction.create({
+            header: transactionHeaderBytes,
+            headerSignature: txnSignature,
+            payload: payloadBytes
+        });
+        return transaction;
+    }
+
+    /**
+     * Builds sawtooth batch from list of small bank transactions
+     * @param {object} args list smallbank transactions
+     * @returns {object} batch list bytes
+     */
+    buildBatch(args) {
+        const {createContext, CryptoFactory} = require('sawtooth-sdk/signing');
+        const context = createContext('secp256k1');
+        const {protobuf} = require('sawtooth-sdk');
+
+        const privateKey = context.newRandomPrivateKey();
+        const signer = new CryptoFactory(context).newSigner(privateKey);
```

Comment:
I think we don't need to new a signer each time when building a batch.

Reply:
I am using one random signing key per batch, it means all transactions are signed with one signing key. In really world scenario each account holder has to sign there transactions to authorize.

Reply:
Ok.